tuf/ngclient/updater.py

```diff
@@ -307,21 +387,34 @@ def _load_root(self) -> None:
         lower_bound = self._trusted_set.root.signed.version + 1
         upper_bound = lower_bound + self.config.max_root_rotations
 
-        for next_version in range(lower_bound, upper_bound):
-            try:
-                data = self._download_metadata(
-                    Root.type,
-                    self.config.root_max_length,
-                    next_version,
-                )
-                self._trusted_set.update_root(data)
-                self._persist_metadata(Root.type, data)
+        save_repo_version_dir = self._spec_version_dir
```

Comment:
I see why you save the original content of `self._spec_version_dir` because you change `self._spec_version_dir` multiple times later, but I had to read it a couple of times.

I suggest instead passing the `version_repo` variable to `download_metadata` and with this you remove:
- `save_repo_version_dir` variable
- `to_break` flag
- the `if` check for `to_break` at the end of the inner cycle and you can directly return after you add to `supported_version_repos` list